recipes/openssl/3.x.x/test_package/conanfile.py

```diff
@@ -1,40 +1,53 @@
 import os
 
+required_conan_version = ">=1.50.2 <1.51.0 || >=1.51.2"
+
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "arch", "build_type"
-    generators = "cmake", "cmake_find_package", "pkg_config"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "VirtualRunEnv"
+    test_type = "explicit"
 
+    @property
     def _with_legacy(self):
-        return (not self.options["openssl"].no_legacy and
-            ((not self.options["openssl"].no_md4) or
-              (not self.options["openssl"].no_rmd160)))
+        openssl = self.dependencies["openssl"]
+        return (not openssl.options.no_legacy and
+            ((not openssl.options.no_md4) or
+              (not openssl.options.no_rmd160)))
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.os == "Android":
+            tc.cache_variables["CONAN_LIBCXX"] = ""
+        openssl = self.dependencies["openssl"]
+        tc.cache_variables["OPENSSL_WITH_ZLIB"] = not openssl.options.no_zlib
+        tc.cache_variables["OPENSSL_WITH_LEGACY"] = self._with_legacy
+        tc.cache_variables["OPENSSL_WITH_MD4"] = not openssl.options.no_md4
+        tc.cache_variables["OPENSSL_WITH_RIPEMD160"] = not openssl.options.no_rmd160
+        tc.generate()
+
 
     def build(self):
         cmake = CMake(self)
-        cmake.definitions["OPENSSL_WITH_ZLIB"] = not self.options["openssl"].no_zlib
-        cmake.definitions["OPENSSL_WITH_LEGACY"] = self._with_legacy()
-        cmake.definitions["OPENSSL_WITH_MD4"] = not self.options["openssl"].no_md4
-        cmake.definitions["OPENSSL_WITH_RIPEMD160"] = not self.options["openssl"].no_rmd160
-        if self.settings.os == "Android":
-            cmake.definitions["CONAN_LIBCXX"] = ""
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not cross_building(self):
-            bin_path = os.path.join("bin", "digest")
-            self.run(bin_path, run_environment=True)
+        if can_run(self):
+            bin_path = os.path.join(self.cpp.build.bindirs[0], "digest")
+            self.run(bin_path, env="conanrun")
 
-            if not self.options["openssl"].no_legacy:
-                bin_legacy_path = os.path.join("bin", "digest_legacy")
+            bin_legacy_path = os.path.join("bin", "digest_legacy")
+            if os.path.exists(bin_legacy_path):
                 self.run(bin_legacy_path, run_environment=True)
-
-            if not self.options["openssl"].no_stdio:
-                self.run("openssl version", run_environment=True)
         assert os.path.exists(os.path.join(self.deps_cpp_info["openssl"].rootpath, "licenses", "LICENSE.txt"))
-
-        for fn in ("libcrypto.pc", "libssl.pc", "openssl.pc",):
-            assert os.path.isfile(os.path.join(self.build_folder, fn))
```

Comment:
Why did you delete this?

Reply:
because these files are no lot generated for the CMakeDeps generator.

 I would need the cmake, cmake_find_package, and pkg_config generators as for the test_v1_package test package (where this check is included).

Also compare that the *.pc files are also only checked for the 1.x.x test_v1_package, not for the 1.x.x test_package.